packages/create-vite/src/index.ts

```diff
@@ -639,6 +657,62 @@ function setupReactSwc(root: string, isTs: boolean) {
       return content.replace('@vitejs/plugin-react', '@vitejs/plugin-react-swc')
     },
   )
+  updateReactCompilerReadme(
+    root,
+    'The React Compiler is currently not compatible with SWC. See [this issue](https://github.com/vitejs/vite-plugin-react/issues/428) for tracking the progress.',
+  )
+}
+
+function setupReactCompiler(root: string, isTs: boolean) {
+  // renovate: datasource=npm depName=babel-plugin-react-compiler
+  const reactCompilerPluginVersion = '19.1.0-rc.3'
+
+  editFile(path.resolve(root, 'package.json'), (content) => {
+    const asObject = JSON.parse(content)
+    const devDepsEntries = Object.entries(asObject.devDependencies)
+    devDepsEntries.push([
+      'babel-plugin-react-compiler',
```

Comment:
one thing that's missing is also using [eslint-plugin-react-hooks@6.0.0-rc.2](https://www.npmjs.com/package/eslint-plugin-react-hooks/v/6.0.0-rc.2). Version 6 will include compiler diagnostics as part of the plugin by default.

Reply:
The react compiler only requires `@babel/types` as a dependency so I'm still unsure what are the need for all these extra dependencies. It adds extra install size (zod 3 & hermes-parser probably not be present for a number of apps) and extra attack surface for package hijacking like what happen yesterday.

Reply:
@ArnaudBarre we inline most dependencies in babel-plugin-react-compiler which is why it only appears to depend on @babel/types (this is mostly a consequence of some Meta-internal infra quirks). For eslint-plugin-react-hooks we didn't, but I can inline as well if it's a hard requirement to get this shipped. Effectively the eslint plugin now includes the entire compiler as it needs to be able to be used standalone, hence the zod (validating compiler configs) and hermes-parser (parsing both Flow and JavaScript/TypeScript into a Babel compatible AST) dependencies.

Reply:
I don't think it's needed to depend on it, it's not directly used: https://github.com/search?q=repo%3Afacebook%2Freact%20path%3A%2F%5Epackages%5C%2Feslint-plugin-react-hooks%5C%2F%2F%20zod&type=code

Instead I think the eslint plugin should make `babel-plugin-react-compiler` a peer-dependency? (maybe optional if it's not required for some rules and lazy loaded for others)